lib/Button.js

```diff
@@ -147,99 +168,107 @@ export default class Button extends Component {
         const textStyle = (() => {
             if (disabled || !(overrides && overrides.textColor)) {
                 return textStyleMap[shape][theme][type];
+            } else {
```

Comment:
I did not remove the `return` statements, I just wrapped them in an `else` to clarify the logic flow.